### PR TITLE
#185754580 Start clean refactor of rules logic in to simpler stores

### PR DESCRIPTION
**Pull request by bakennedy**
Rules lists for not/matching/unidentified checking of every entity rules and cohort membership updates

### CODE REVIEW

**Comment by bakennedy** [comment]
This has been tested against:
matching - in-cohort
matching - not-in-cohort
not_matching - in-cohort
not_matching - not-in-cohort
matching - unidentified
not_matching - unidentified
regex matching
regex not-matching

all of the test cases succeed and apply or not apply rules as expected, and the rule order of operation remains correct

File: governance_rules.go [hunk 1]
```diff
@@ -119,24 +124,55 @@ type TemplatedOverrideValues struct {
...
+	// and simply check if the current entity is in any cohort for a given rule to apply non_matching rules
+	isInCohort, matching := GetMatchingRuleTemplates(config, userValues)
+	regexToCheck = append(regexToCheck, matching...)
+	regexToCheck = append(regexToCheck, GetNotMatchingRuleTemplates(config.UserRules, userId, isInCohort)...)
+
+	isInCohort, matching = GetMatchingRuleTemplates(config, companyValues)
+	regexToCheck = append(regexToCheck, matching...)
+	regexToCheck = append(regexToCheck, GetNotMatchingRuleTemplates(config.CompanyRules, companyId, isInCohort)...)
+
+	for _, r := range config.Regex {
```
Comment (dgilling) [inline governance_rules.go:167]: @bakennedy is this ordering correct such that user is highest pri by being overridden last? It seems to not match the comments above.

Comment (bakennedy) [reply governance_rules.go:167]: Did I not re-write some comments?  I'll confirm

Comment (bakennedy) [reply governance_rules.go:167]: Yep, that's the original commentary on this code path, and fixed

**Review by dgilling — APPROVED** [review]

**Pull request merged**